apps/veil/src/pages/trade/ui/order-form/store/MarketOrderFormStore.ts

```diff
@@ -123,7 +126,9 @@ export class MarketOrderFormStore {
   };
 
   get quoteInput(): string {
-    return this._quoteAssetInput;
+    return this._quoteAssetInput.length
+      ? round({ value: Number(this._quoteAssetInput), decimals: this._quoteAsset?.exponent ?? 6 })
+      : '';
   }
```

Comment:
I think the store should keep the field as is, and the component should be responsible for displaying it.